build-info-extractor/src/main/java/org/jfrog/build/extractor/BuildInfoExtractorUtils.java

```diff
@@ -332,4 +332,83 @@ private static void addPropsFromCommandSystemProp(Properties additionalProps, Lo
             }
         }
     }
+
+    /**
+     * Creates a build info link to the published build. This method is in use also in the Jenkins Artifactory plugin.
+     *
+     * @param url         JFrog Platform or Artifactory URL
+     * @param buildName   Build name of the published build
+     * @param buildNumber Build number of the published build
+     * @param timeStamp   Timestamp (started date time in milliseconds) of the published build
+     * @param project     Project of the published build
+     * @param encode      True if should encode build name and build number
+     * @param platformUrl True if the input url is platform's URL, false if it is Artifactory's URL
+     * @return Link to the published build.
+     */
+    public static String createBuildInfoUrl(String url, String buildName, String buildNumber, String timeStamp,
+                                            String project, boolean encode, boolean platformUrl) {
+        if (platformUrl) {
+            // Platform URL provided. We have everything we need to create the build info URL.
+            return createBuildInfoUrl(url, buildName, buildNumber, timeStamp, project, encode);
+        }
+        if (isNotBlank(project)) {
+            if (endsWith(url, "/artifactory")) {
+                // If the project parameter provided, try to guess the platform URL.
+                return createBuildInfoUrl(removeEnd(url, "/artifactory"), buildName,
+                        buildNumber, timeStamp, project, encode);
+            }
+            // If Artifactory's URL doesn't end with "/artifactory", it is impossible to create the URL.
+            return "";
```

Comment:
Wouldn't it be better to throw an exception here?

Reply:
No. This is part of a perfectly legal flow. The build info URL can't be determined in this very specific case. In that case, we may prefer to not display it.
Also, this function is in use in many places and we may not like to add a try-catch mechanism for each one of them.